code/iaas/auth-logic/src/main/java/io/cattle/platform/iaas/api/auth/integration/ldap/LdapIdentitySearchProvider.java

```diff
@@ -443,4 +462,30 @@ private String escapeLDAPSearchFilter(String filter) {
         }
         return sb.toString();
     }
+
+    @PostConstruct
+    public void init() {
+        if (contextPool == null) {
+            GenericObjectPoolConfig config = new GenericObjectPoolConfig();
+            contextPool = new GenericObjectPool<>(new LdapServiceContextPoolFactory(), config);
+        }
+    }
+
+    private LdapContext getServiceContext() {
+        try {
+            return contextPool.borrowObject();
+        } catch (Exception e) {
+            logger.error("Failed to get service context for ldap.", e);
+            return null;
+        }
+    }
+
+    @Inject
+    public void setExecutorService(ExecutorService executorService) {
```

Comment:
Nit: instead of using getters/setters, you can just annotate the service you are trying to inject, in its declaration with @ Inject.